setup.py

```diff
@@ -1,49 +1,21 @@
 # -*- coding: utf-8 -*-
 
-from setuptools import setup
 import os
 import sys
-import platform
-try:
-    from urllib.request import urlretrieve
-except ImportError:
-    from urllib import urlretrieve
 
-long_description = open("README.rst").read()
-with open('requirements.txt', 'r') as fp:
+from setuptools import setup
+
+python_version_major = sys.version_info.major
+
+with open("README.rst") as f:
+    long_description = f.read()
+with open('requirements.txt') as fp:
     install_requires = fp.read()
 extras_require = {}
 for extra in ['extras', 'docs', 'tests']:
-    with open('requirements-{0}.txt'.format(extra), 'r') as fp:
+    with open('requirements-{0}.txt'.format(extra)) as fp:
         extras_require[extra] = fp.read()
 
-# spade specific
```

Comment:
If I understand correctly, the `fim.so` is now downloaded whenever the `spade` module first is imported. 

Nevertheless, when installing the elephant package from this branch (using `python setup.py install`), `fim.so` will already reside in the install directory automatically even before importing `spade`. This is in my opinion the desired behavior, since a user may not have write privileges to the install folder after installation, e.g. on a system wide install. 

However, I am not sure why the download is actually triggered during install, since the corresponding lines are removed from `setup.py`.

Is this potentially due to the massive importing of all modules in `__init.py__`, which triggers the download during install? If so, is #231 a problem, which removes these imports?

Reply:
The download now is not triggered during the install. I did it because travis does not download fim during the install (travis issue only). But I see the point - it's better to download the file immediately. And if we're about to get rid of pythonic fim implementation, we need to force travis to download this file to maintain the fim tests properly. I'll think how to do this